HotelBookingTests/HotelBaseClassTest.cs

```diff
@@ -1,4 +1,7 @@
-﻿using HotelBase;
+﻿using System;
```

Comment:
Should have:

Somewhere in this test class is ShouldValidateEmail method. It is my mistake that I have missed that at first so sorry fot that but:
* Instead of having one test I will advise to have to simpler tests:
  * WhenEmailContainsForbiddenCharacters_ShouldNotAllowToSendIt
  * WhenEmailIsCorrect_ShouldAllowToSentIt

Reply:
Applied